cmd/umserver/main_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"database/sql"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/connect"
+	"github.com/segmentio/kafka-go"
+)
+
+type MockSetupComponents struct {
```

Comment:
You can generate mocks using `github.com/golang/mock` 

Or write own implementation, for example:
```
func NewCloserMock(name string, expected func() error) io.Closer {
	return CloserMock{
		name: name,
		expected: expected,
	}
}

type CloserMock struct {
	name     string
	expected func() error
}

func (cm CloserMock) Close() error {
	return cm.expected()
}
```

Usage:
```

	closers := []io.Closer{
		NewCloserMock("consul", func() error {
			return nil
		}),
		NewCloserMock("postgres", func() error {
			return errors.New("can't close connection")
		}),
	}
```

Reply:
I really appreciate your advice. I used it